packages/platform-browser/src/browser.ts

```diff
@@ -88,6 +89,7 @@ export const BROWSER_MODULE_PROVIDERS: StaticProvider[] = [
   {provide: DomSharedStylesHost, useClass: DomSharedStylesHost, deps: [DOCUMENT]},
   {provide: Testability, useClass: Testability, deps: [NgZone]},
   {provide: EventManager, useClass: EventManager, deps: [EVENT_MANAGER_PLUGINS, NgZone]},
+  {provide: XhrFactory, useClass: BrowserXhr},
```

Comment:
I believe this would allow us to continue using `StaticProvider[]`:

```suggestion
  {provide: XhrFactory, useClass: BrowserXhr, deps: []},
```

Reply:
Isn’t `deps: []` the same as when it’s omitted?

Reply:
For `StaticProvider` it is required:

https://github.com/angular/angular/blob/cbba7152fd2a533277b88aca98a50339d098d9f7/packages/core/src/di/interface/provider.ts#L66-L70